### PR TITLE
Rviz StartDetectPanel

### PR DESCRIPTION
**Pull request by JoshMJ37**
Made StartDetectPanel for rviz to tell us when the stat-detected node is publishing and true, publishing and false, or not publishing.

### CODE REVIEW

File: rr_rviz_plugins/plugin_description.xml [hunk 1]
```diff
@@ -13,4 +13,11 @@
             Panel used for reseting the state to race start
         </description>
     </class>
-</library>
\ No newline at end of file
+    <class name="rr_rviz_plugins/StartDetectPanel"
+           type="rr_rviz_plugins::StartDetectPanel"
+           base_class_type="rviz::Panel">
+        <description>
+            Panel used for reseting the state to race start
```
[the remark this answers is not in the document]
Comment (JoshMJ37) [reply rr_rviz_plugins/plugin_description.xml]: All this panel is supposed to do is let the user know whether or not the StartDetect publisher has started publishing and then whether or not it is publishing true or false.

File: rr_rviz_plugins/include/rr_rviz_plugins/StartDetectPanel.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     * the lifetime of the widget.
+     */
+    // ros::Subscriber startDetect_subscriber;
+    ros::Subscriber startDetect_subscriber;
+
+    /**
+     * Declare any ROS callbacks you need here. Be sure to create a parameter for any UI elements you need to update.
+     * @param msg The ROS message that triggers this callback.
+     * @param label A QT label whose text we will update based on the message contents.
+     */
```
Comment (ebretl) [inline rr_rviz_plugins/include/rr_rviz_plugins/StartDetectPanel.h]: All the generic comments like this should be removed or changed to be more meaningful

**Comment by btdubs** [comment]
I'd like to request that you add a visual indicator (some green go light/sign and a red stop light/sign) for when the start is detected and not detected respectively.

Also, it might be nice to merge the reset start detected button into this panel so all of our start information is in one place.

Otherwise, nice job guys.

**Comment by ebretl** [comment]
@JoshMJ37 @brandoDecu This has been sitting on "changes requested" for a while.... Are you going to revisit this, or would you like someone else to take care of it?

**Comment by brandoDecu** [comment]
Hi Evan,

Sorry for not having shown up the past 2 weeks. It's fine for me if someone
else wants to work on it.

I wish you a good break.

Brando

**Pull request closed without merging**